### PR TITLE
[#IOCOM-98] Migrate node 18

### PR DESCRIPTION
**Pull request by AndreaPagoPa**
#### List of Changes

- [node upgrade to 18.13.0](https://github.com/pagopa/io-functions-app/commit/144753fbb55e15d2ea2848750a9b8abf5b092cc3)

#### Motivation and Context
End of life Node 14.

#### How Has This Been Tested?
Build

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: __integrations__/docker-compose.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: "3.2"
 services:
 
   functions-node-14:
```
Comment (Garma00) [inline __integrations__/docker-compose.yml:5]:
```suggestion
  functions-node-18:
```

File: __integrations__/docker-compose.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ version: "3.2"
...
-    image: functions-node-14
+    image: functions-node-18
     build:
       context: ..
       dockerfile: ./docker/functions-node-14/Dockerfile
```
Comment (Garma00) [inline __integrations__/docker-compose.yml:9]:
```suggestion
      dockerfile: ./docker/functions-node-18/Dockerfile
```

File: docker/functions-node-14/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM node:14.16.0
+FROM node:18.13.0
```
Comment (Garma00) [inline docker/functions-node-14/Dockerfile:1]: Rename the directory as well `functions-node-18/`

File: __integrations__/docker-compose.yml (hunk 2 above)
Comment (Garma00) [reply __integrations__/docker-compose.yml:9]: Also check out the `functions-node-14` under `function` `depends_on`

File: package.json [hunk 4]
```diff
@@ -58,8 +59,8 @@
     "@azure/cosmos": "^3.17.1",
     "@azure/storage-queue": "^12.4.0",
     "@pagopa/express-azure-functions": "^2.0.0",
```
Comment (michaeldisaro) [inline package.json:61]: We have to update this to the new version.

**Review by michaeldisaro — APPROVED** [review]
lgtm

**Pull request merged**